addon/components/x-tree-node.js

```diff
@@ -32,10 +56,24 @@ export default Component.extend({
     }
   },
 
+  setChildCheckboxesRecursively(parentNode, checkValue) {
+    const children = parentNode.children;
+    if (children.length) {
+      children.setEach('isChecked', checkValue);
+      children.forEach((child) => {
```

Comment:
`child` doesn't need the parenthesis and maybe the two operations can be combined:
```
children.forEach(child => {
  child.set('isChecked', checkValue);
  this.setChildCheckboxesRecursively(child, checkValue);
});
```